src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,847 @@
+                    !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesRuntimeInformation, out var runtimeInformationType))
+                {
+                    return;
+                }
+
+                var guardMethods = GetRuntimePlatformGuardMethods(runtimeInformationType, operatingSystemType!);
+
+                context.RegisterOperationBlockStartAction(context => AnalyzeOperationBlock(context, guardMethods, osPlatformType));
+            });
+
+            static ImmutableArray<IMethodSymbol> GetRuntimePlatformGuardMethods(INamedTypeSymbol runtimeInformationType, INamedTypeSymbol operatingSystemType)
+            {
+                return operatingSystemType.GetMembers().OfType<IMethodSymbol>().Where(m => s_platformCheckMethodNames.Contains(m.Name)).ToImmutableArray().
+                    Add(runtimeInformationType.GetMembers().OfType<IMethodSymbol>().Where(m => IsOSPlatform == m.Name).FirstOrDefault());
+            }
+        }
+
+        private void AnalyzeOperationBlock(OperationBlockStartAnalysisContext context, ImmutableArray<IMethodSymbol> guardMethods, INamedTypeSymbol osPlatformType)
+        {
+            var platformSpecificOperations = PooledConcurrentDictionary<IOperation, SmallDictionary<string, PlatformAttributes>>.GetInstance();
+            var platformSpecificMembers = PooledConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>>.GetInstance();
+
+            context.RegisterOperationAction(context =>
+            {
+                AnalyzeOperation(context.Operation, context, platformSpecificOperations, platformSpecificMembers);
+            },
+            OperationKind.MethodReference,
+            OperationKind.EventReference,
+            OperationKind.FieldReference,
+            OperationKind.Invocation,
+            OperationKind.ObjectCreation,
+            OperationKind.PropertyReference);
+
+            context.RegisterOperationBlockEndAction(context =>
+            {
+                try
+                {
+                    if (platformSpecificOperations.IsEmpty)
+                    {
+                        return;
+                    }
+
+                    if (guardMethods.IsEmpty || !(context.OperationBlocks.GetControlFlowGraph() is { } cfg))
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
+                        cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
+                        context.Options, RequiredOsRule, performValueContentAnalysis: true,
```

Comment:
Force executing ValueContent dataflow analysis every time is likely one of the biggest contributors to this analyzer being expensive and slow. We should really restore the logic to execute this expensive analysis only if required - given that majority of times the guard methods are called directly inside the if condition, value content analysis is just redundant for all those cases.

Reply:
Thank you for resolving this